int32 operation_num
int32[] prev_ops
int32[] durations # size should be equal to size of machines, if machine cannot do it = -1
